### PR TITLE
Inspection batch with failure

### PR DESCRIPTION
**Pull request by CermakM**
Add inspection-batch-with-failure workflow example

This workflow submits a Job which creates `n` pod replicas and runs them in a sequence
The pod which is being ran is faulty and fails. The workflow is supposed to wait untill
a certain amount of job failures appear and fail as well.

successCondition: `n` pods have finished successfully and are present in the cluster
failureCondition: there are more failures than is allowed (a constant to account for potential flakes)

The workflow simulates the scenario when we want to create a batch of inspections but the stack is faulty
and we don't want to pollute the namespace unnecessarily with failed pods.

### CODE REVIEW

**Comment by CermakM** [comment]
[Create examples of Argo Workflows.](https://trello.com/c/rbSVX7o4/730-create-examples-of-argo-workflows)

File: examples/inspection-batch-with-failure.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          completions: {{workflow.parameters.batch_size}}
+          template:         
+            metadata:
+              name: "{{workflow.name}}"
+              labels:
+                mark: cleanup
+            spec:
+              containers:
+              - name: fake-inspection-with-failure
+                image: alpine:latest
```
Comment (goern) [inline examples/inspection-batch-with-failure.yaml]: ubi8!

File: examples/inspection-batch-with-failure.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  - name: stop
+    resource:
+      action: patch
+      manifest: |
+        apiVersion: batch/v1
+        kind: Job
+        metadata:
+          name: "{{workflow.name}}"
+        spec:
+          parallelism: 0
```
Comment (goern) [inline examples/inspection-batch-with-failure.yaml]: missing \n

**Review by goern — CHANGES_REQUESTED** [review]

**Review by goern — COMMENTED** [review]
this needs some coala refinement yamllint config style?!

**Comment by CermakM** [comment]
recheck

**Review by goern — APPROVED** [review]

**Pull request merged**